### ISSUE TITLE
Envoy-S Metered reports the wrong production values when the current transformers are not installed

### ISSUE THREAD

**Issue by WesleyE** [pyenphase/pyenphase#99]
**Describe the bug**
I have an Envoy-S Metered where they unfortunately did not install the current transformers. This results in the `whToday` and `whLifetime` to not be zero, causing the code in system_production.py@47 to report the wrong values.

If you have pointers, I'm willing to solve the issue. Can I use the eim `activeCount` to also conditionally load the `whLifetime`?

**To Reproduce**
- Have Envoy-S Metered, current transformers not installed
- See eim values being non-zero in production.json
- See `whLifetime` output with the eim value instead of the inverters

**Additional context**
production.json values:
```json
{
    "production": [
        {
            "type": "inverters",
            "activeCount": 6,
            "readingTime": 1697119478,
            "wNow": 16,
            "whLifetime": 15212
        },
        {
            "type": "eim",
            "activeCount": 0,
            "measurementType": "production",
            "readingTime": 1697119517,
            "wNow": 0,
            "whLifetime": 0.002,
            "varhLeadLifetime": 0,
            "varhLagLifetime": 0.002,
            "vahLifetime": 0,
            "rmsCurrent": 0,
            "rmsVoltage": 232.121,
            "reactPwr": 0,
            "apprntPwr": 0,
            "pwrFactor": 1,
            "whToday": 0.002,
            "whLastSevenDays": 0.002,
            "vahToday": 0,
            "varhLeadToday": 0,
            "varhLagToday": 0.002
        }
    ],
    "consumption": [
        {
            "type": "eim",
            "activeCount": 0,
            "measurementType": "total-consumption",
            "readingTime": 1697119517,
            "wNow": -1.007,
            "whLifetime": 0.002,
            "varhLeadLifetime": 0,
            "varhLagLifetime": -0.004,
            "vahLifetime": 0,
            "rmsCurrent": 0.022,
            "rmsVoltage": 247.834,
            "reactPwr": 0,
            "apprntPwr": 0.107,
            "pwrFactor": -1,
            "whToday": 0.002,
            "whLastSevenDays": 0.002,
            "vahToday": 0,
            "varhLeadToday": 0,
            "varhLagToday": 0
        },
        {
            "type": "eim",
            "activeCount": 0,
            "measurementType": "net-consumption",
            "readingTime": 1697119517,
            "wNow": -0.987,
            "whLifetime": 0,
            "varhLeadLifetime": 0,
            "varhLagLifetime": 0,
            "vahLifetime": 0,
            "rmsCurrent": -0.269,
            "rmsVoltage": 247.834,
            "reactPwr": 0,
            "apprntPwr": -2.134,
            "pwrFactor": -1,
            "whToday": 0,
            "whLastSevenDays": 0,
            "vahToday": 0,
            "varhLeadToday": 0,
            "varhLagToday": 0
        }
    ],
    "storage": [
        {
            "type": "acb",
            "activeCount": 0,
            "readingTime": 0,
            "wNow": 0,
            "whNow": 0,
            "state": "idle"
        }
    ]
}
```

**Comment by catsmanac** [comment]
This is a known issue with Envoy Metered without CT configured. whLifetime needs to come from inverters section in production.json. So for watt_hours_lifetime use `now_source["whLifetime"]` as for watts_now. (Wh today and Wh Last 7 days will not produce usefull numbers).

```
        # This is backwards compatible with envoy_reader
        now_source = eim if eim["activeCount"] else inverters

        return cls(
            watt_hours_lifetime=int(
                round(eim.get("whLifetime") or inverters.get("whLifetime") or 0)
            ),
            watt_hours_last_7_days=int(
                round(
                    eim.get("whLastSevenDays") or inverters.get("whLastSevenDays") or 0
                )
            ),
            watt_hours_today=int(
                round(eim.get("whToday") or inverters.get("whToday") or 0)
            ),
            watts_now=int(round(now_source["wNow"])),
        )
```

**Comment by catsmanac** [comment]
When looking at the issue and considering the data in HA Issue [101354](https://github.com/home-assistant/core/issues/101354#issuecomment-1791317304) reported by @ChrisCrewdson it looks that the reply of /api/v1/production is intervening with the fallback to inverters data from the /production endpoint. Since the envoy returns 1 non-zero value the fallback is not  triggered. Not sure if that is the case with @WesleyE as well or if that issue only needs [above change](https://github.com/pyenphase/pyenphase/issues/99#issuecomment-1760174519)

Tthe change of using `now_source` for lifetime is needed but will not work on it's own for HA issue 101354. The change done in #83 added test for all zero's as reported for fw 8 in #74

```python
        if all(value == 0 for value in response.values()):
            _LOGGER.debug(
                "Detected broken production endpoint bug at %s: %s",
                URL_PRODUCTION_V1,
                response,
            )
            return None
```
Probably enabling `EnvoyProductionJsonFallbackUpdater` if all 3 wattHour values are 0, ignoring wattsNow, may solve this issue.

```JSON
{
  "wattHoursToday": 0,
  "wattHoursSevenDays": 0,
  "wattHoursLifetime": 0,
  "wattsNow": -130
}
```

**Comment by WesleyE** [comment]
I'm on firmware `D7.6.172 (94a00d)`. I've tried to contribute back, but could not warp my head around how to properly add new testcases in the limited amount of time I had. Will try again if not solved by your changes in a few days.

**/api/v1/production:**
```json
{
  "wattHoursToday": 0,
  "wattHoursSevenDays": 0,
  "wattHoursLifetime": 0,
  "wattsNow": 0
}
```

**/ivp/meters:**
```json
[
  {
    "eid": 704643328,
    "state": "disabled",
    "measurementType": "production",
    "phaseMode": "three",
    "phaseCount": 1,
    "meteringStatus": "normal",
    "statusFlags": [
      
    ]
  },
  {
    "eid": 704643584,
    "state": "disabled",
    "measurementType": "net-consumption",
    "phaseMode": "three",
    "phaseCount": 3,
    "meteringStatus": "normal",
    "statusFlags": [
      
    ]
  }
]
```

**Comment by catsmanac** [comment]
The fix will not solve the issue for whToday and whLastSevenDays, only for whLifetime. Envoy is not sending correct values for these in this configuration as far as I know. You'll have to use utilitymeters of whLifetime in HA if you need these.

**Comment by bdraco** [comment]
fixed via #111
